web/courses.go

```diff
@@ -333,30 +328,35 @@ func (s *QuickFeedService) getEnrollmentsWithActivity(courseID uint64) ([]*qf.En
 	if err != nil {
 		return nil, err
 	}
+	// fetch course record with all assignments and active enrollments
+	course, err := s.db.GetCourse(courseID, true)
+	if err != nil {
+		return nil, err
+	}
 	var enrollmentsWithActivity []*qf.Enrollment
-	for _, enrolLink := range allEnrollmentsWithSubmissions.Links {
-		enrol := enrolLink.Enrollment
+	for _, enrollment := range course.Enrollments {
 		var totalApproved uint64
 		var submissionDate time.Time
-		for _, submissionLink := range enrolLink.Submissions {
-			submission := submissionLink.Submission
-			if submission != nil {
-				if submission.Status == qf.Submission_APPROVED {
-					totalApproved++
-				}
-				if enrol.LastActivityDate == "" {
-					submissionDate, err = submission.NewestSubmissionDate(submissionDate)
-					if err != nil {
-						return nil, err
+		if submissions, ok := submissionsMap.Submissions[enrollment.ID]; ok {
```

Comment:
You can add this snippet in the `qf` package:
```go
// For returns the course's submissions for the given ID.
func (cs *CourseSubmissions) For(id uint64) []*Submission {
	submissions := cs.GetSubmissions()
	if submissions == nil {
		return nil
	}
	return submissions[id].GetSubmissions()
}
```
And simplify this code like this:
```go
for _, submission := range submissionsMap.For(enrollment.ID) {
	if submission.Status == qf.Submission_APPROVED {
		totalApproved++
	}
...
```
Maybe `submissions` would then be a better name instead of `submissionsMap`